location goal
---
int64 result
